NTupleAnalyzer_nanoAOD.py

```diff
@@ -2210,11 +2224,22 @@ def GetLLJJMasses3Jets(l1,l2,j1,j2,j3):
 	Branches['pass_HLTTkMu50'][0]      = t.HLT_TkMu50
 
 	#fixme some flags missing, others not working correctly
-	Branches['Flag_BadChargedCandidateFilter'][0]         = t.Flag_BadChargedCandidateFilter		
-	Branches['Flag_BadChargedCandidateSummer16Filter'][0] = 1#t.Flag_BadChargedCandidateSummer16Filter 
-	Branches['Flag_BadGlobalMuon'][0]	       	      = t.Flag_BadGlobalMuon			
-	Branches['Flag_BadPFMuonFilter'][0]	       	      = t.Flag_BadPFMuonFilter			
-	Branches['Flag_BadPFMuonSummer16Filter'][0]    	      = 1#t.Flag_BadPFMuonSummer16Filter  		
+	if 'SingleMuon' in name or 'SingleElectron' in name or 'DoubleMuon' in name or 'DoubleEG' in name:
+		Branches['Flag_BadChargedCandidateFilter'][0]         = t.Flag_BadChargedCandidateFilter		
+		Branches['Flag_BadChargedCandidateSummer16Filter'][0] = t.Flag_BadChargedCandidateSummer16Filter 
+		Branches['Flag_BadGlobalMuon'][0]	       	      = 1#t.Flag_BadGlobalMuon			
+		Branches['Flag_BadPFMuonFilter'][0]	       	      = t.Flag_BadPFMuonFilter			
+		Branches['Flag_BadPFMuonSummer16Filter'][0]    	      = t.Flag_BadPFMuonSummer16Filter  
+		Branches['Flag_ecalBadCalibFilter'][0]                = t.Flag_ecalBadCalibFilter 
+		Branches['passPrimaryVertex'][0]                      = t.Flag_goodVertices
+	else:
+		Branches['Flag_BadChargedCandidateFilter'][0]         = 1		
```

Comment:
Please at least put back the old names, but commented out. Looking at /eos/cms/store/group/phys_exotica/leptonsPlusJets/LQ/customNano/gmadigan/DY1JetsToLL_M-50_LHEZpT_150-250_TuneCP5_13TeV-amcnloFXFX-pythia8/LQ-v1/190819_141334/0000/DY1JetsToLL_M-50_LHEZpT_150-250_ext2_1.root , I see many of these filters in MC, so I don't understand why they are being set to 1
<img width="245" alt="Screen Shot 2019-09-26 at 22 06 32" src="https://user-images.githubusercontent.com/4973302/65721213-e7872780-e0a9-11e9-826c-b5c123c50855.png">

Reply:
That path is actually not for the skimmed files. If you look at a skimmed LQ sample, e.g., /eos/cms/store/group/phys_exotica/leptonsPlusJets/LQ/customNano/skim/LQ2/LQToCMu_M-1000_BetaOne_pythia8/LQToCMu_M-1000_BetaOne_pythia8_tree_0.root you'll see a few missing flags (e.g., Flag_BadChargedCandidateFilter). What should the branches in the ntuple analyzer be filled with in the case that it doesn't exist in the skimmed file?

<img width="536" alt="Screen Shot 2019-09-27 at 10 21 48 AM" src="https://user-images.githubusercontent.com/11445369/65776767-c0177580-e110-11e9-993a-ac54f99331ac.png">

Reply:
I do see it there - though it is not with the other flags. Please try to figure out why it is failing then if it is indeed in the file.

<img width="281" alt="Screen Shot 2019-09-27 at 20 28 31" src="https://user-images.githubusercontent.com/4973302/65793179-89b81580-e165-11e9-952e-55b75a115056.png">